server/bin/pbench-tarball-stats

```diff
@@ -0,0 +1,193 @@
+#!/usr/bin/env python3
+# -*- mode: python -*-
+
+"""Pbench Tar Ball Stats
+
+Scan through the ARCHIVE hierarchy for tar balls and report some
+statistics on them:
+
+  * Total number of controllers
+  * Tar balls per controller (most to least)
+  * Tar balls by month for all years, count
+  * Tar balls by month for all years, total size
+  * Top 10 most recent controllers
+  * Top 10 largest tar balls
+"""
+
+import collections
+import locale
+import os
+import re
+import sys
+
+from datetime import datetime
+from argparse import ArgumentParser
+
+import pbench
+from pbench import (
+    PbenchConfig,
+    BadConfig,
+    get_pbench_logger,
+)
+
+
+_NAME_ = "pbench-tarball-stats"
+
+tb_pat_r = (
+    r"\S+_(\d\d\d\d)[._-](\d\d)[._-](\d\d)[T_](\d\d)[._:](\d\d)[._:](\d\d)\.tar\.xz"
+)
+tb_pat = re.compile(tb_pat_r)
+
+
+def gen_tar_balls(archive):
+    """gen_tar_balls - traverse the given ARCHIVE hierarchy looking tar balls,
+    returning information about the tar so that statistics can be generated.
+    """
+    with os.scandir(archive) as archive_scan:
+        for c_entry in archive_scan:
+            if c_entry.name.startswith(".") and c_entry.is_dir(follow_symlinks=False):
+                continue
+            if not c_entry.is_dir(follow_symlinks=False):
+                # NOTE: the pbench-audit-server should pick up and flag this
+                # unwanted condition.
+                continue
```

Comment:
So we skip `.*` directory files and all non-directory files. I think these two statements could be safely consolidated to

```
if c_entry.name.startswith(".") or not c_entry.is_dir(follow_symlinks=False):
    continue
````